File: hashen_bitproject_1800135_new/server/src/main/java/bit/project/server/entity/Taskallocation.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import java.time.LocalDate;
import java.time.LocalTime;
import javax.persistence.*;
import javax.persistence.Id;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Taskallocation{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDate date;

    private LocalTime time;

    private String pobox;

    private String street;

    @Lob
    private String remarks;

    @Lob
    private String title;

    private LocalDateTime tocreation;


    @ManyToOne
    private Tasktype tasktype;

    @ManyToOne
    private Gramaniladharidiv gramaniladharidiv;

    @ManyToOne
    private Connectionrequest connectionrequest;

    @ManyToOne
    private Disconnectionrequest disconnectionrequest;

    @ManyToOne
    private Reconnectionrequest reconnectionrequest;

    @ManyToOne
    private Modificationrequest modificationrequest;

    @ManyToOne
    private Complaint complaint;

    @ManyToOne
    private Taskallocationstatus taskallocationstatus;

    @OneToMany(mappedBy="taskallocation", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Taskallocationitem> taskallocationitemList;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    @ManyToMany
        @JoinTable(
        name="taskallocationvehicle",
        joinColumns=@JoinColumn(name="taskallocation_id", referencedColumnName="id"),
        inverseJoinColumns=@JoinColumn(name="vehicle_id", referencedColumnName="id")
    )
    private List<Vehicle> vehicleList;

    @ManyToMany
        @JoinTable(
        name="taskallocationemployee",
        joinColumns=@JoinColumn(name="taskallocation_id", referencedColumnName="id"),
        inverseJoinColumns=@JoinColumn(name="employee_id", referencedColumnName="id")
    )
    private List<Employee> employeeList;


    public Taskallocation(Integer id){
        this.id = id;
    }

    public Taskallocation(Integer id, String code, Gramaniladharidiv gramaniladharidiv, Connectionrequest connectionrequest, Taskallocationstatus taskallocationstatus){
        this.id = id;
        this.code = code;
        this.gramaniladharidiv = gramaniladharidiv;
        this.connectionrequest = connectionrequest;
        this.taskallocationstatus = taskallocationstatus;
    }

}

File: hashen_bitproject_1800135_new/client/src/app/entities/taskallocation.ts
import {User} from './user';
import {Vehicle} from './vehicle';
import {Tasktype} from './tasktype';
import {Employee} from './employee';
import {Complaint} from './complaint';
import {DataPage} from '../shared/data-page';
import {Gramaniladharidiv} from './gramaniladharidiv';
import {Connectionrequest} from './connectionrequest';
import {Taskallocationitem} from './taskallocationitem';
import {Reconnectionrequest} from './reconnectionrequest';
import {Modificationrequest} from './modificationrequest';
import {Disconnectionrequest} from './disconnectionrequest';
import {Taskallocationstatus} from './taskallocationstatus';

export class Taskallocation {
  id: number;
  code: string;
  date: string;
  time: string;
  tasktype: Tasktype;
  pobox: string;
  //add title to EEntity
  title: string;
  street: string;
  gramaniladharidiv: Gramaniladharidiv;
  connectionrequest: Connectionrequest;
  disconnectionrequest: Disconnectionrequest;
  reconnectionrequest: Reconnectionrequest;
  modificationrequest: Modificationrequest;
  complaint: Complaint;
  taskallocationstatus: Taskallocationstatus;
  remarks: string;
  taskallocationitemList: Taskallocationitem[];
  vehicleList: Vehicle[];
  employeeList: Employee[];
  creator: User;
  tocreation: string;
}

export class TaskallocationDataPage extends DataPage{
    content: Taskallocation[];
}
